lib/internal/process/next_tick.js

```diff
@@ -5,6 +5,7 @@ exports.setup = setupNextTick;
 function setupNextTick() {
   const promises = require('internal/process/promises');
   const emitPendingUnhandledRejections = promises.setup(scheduleMicrotasks);
+  const MAX_TICKS = 1e4;
```

Comment:
@addaleax's interpretation is correct at a base level. I'm not sure if breaking out in this way runs more nextTicks until the next turn of the event loop or if it runs another loop if more nextTicks remain/are scheduled.